File: Week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    double L = (double) letters / words * 100.0;
    double S = (double) sentences / words * 100.0;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    int roundIndex = index + 0.5;
    // Print the grade level

    if (roundIndex < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (roundIndex >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", roundIndex);
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_words(string text)
{
    // Return the number of words in text
    int count = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count += 1;
        }
    }
    return count;
}

File: Week6/sentimental-readability/readability.py
from cs50 import get_string


def count_letters(text):
    count = 0
    for cha in text:
        if cha.isalpha():
            count += 1

    return count


def count_words(text):
    count = 1
    for cha in text:
        if cha.isspace():
            count += 1

    return count


def count_sentences(text):
    count = 0
    for cha in text:
        if cha in [".", "!", "?"]:
            count += 1

    return count


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if (index < 1):
        print("Before Grade 1\n")

    elif (index >= 16):
        print("Grade 16+\n")

    else:
        print(f"Grade {index}")


main()
